### PR TITLE
DDF-2442 removed use of DTD in XML parsing

### PR DESCRIPTION
**Pull request by stustison**
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@bdeining @troymohl 
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@jlcsmith
@stustison
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

**Comment by bdeining** [comment]
:+1:

File: catalog/transformer/catalog-transformer-mgmp/src/main/java/org/codice/alliance/catalog/transformer/mgmp/MgmpTransformer.java [hunk 1]
```diff
@@ -486,6 +486,16 @@ private void addMetacardDataQuality(MetacardImpl metacard) {
         try (InputStream inputStream = getSourceInputStream()) {
             DocumentBuilderFactory domFactory = DocumentBuilderFactory.newInstance();
             domFactory.setNamespaceAware(false);
+            try {
+                domFactory.setFeature(
+                        "http://apache.org/xml/features/nonvalidating/load-dtd-grammar",
+                        false);
+                domFactory.setFeature(
+                        "http://apache.org/xml/features/nonvalidating/load-external-dtd",
+                        false);
```
Comment (troymohl) [inline catalog/transformer/catalog-transformer-mgmp/src/main/java/org/codice/alliance/catalog/transformer/mgmp/MgmpTransformer.java]: Per https://xerces.apache.org/xerces-j/features.html this seems to indicate that these features can only be disabled if validation is disabled. Does that mean we are not validating the incoming data and just handing it over to the parser?

Comment (stustison) [inline catalog/transformer/catalog-transformer-mgmp/src/main/java/org/codice/alliance/catalog/transformer/mgmp/MgmpTransformer.java]: I was just following the owasp recommendations, which only include disabling these features.

Comment (jlcsmith) [inline catalog/transformer/catalog-transformer-mgmp/src/main/java/org/codice/alliance/catalog/transformer/mgmp/MgmpTransformer.java]: I may be wrong, but I don't think we would ever be dealing with DTD xml, so I think this is OK to disable.

Comment (stustison) [inline catalog/transformer/catalog-transformer-mgmp/src/main/java/org/codice/alliance/catalog/transformer/mgmp/MgmpTransformer.java]: Shouldn't be, but leaving DTDs on leaves you open to many different attacks.

**Comment by jlcsmith** [comment]
Built successfully.  LGTM 👍

**Pull request merged**